woocommerce/woocommerce
# Upgrade to pnpm 7

### All Submissions:






### Changes proposed in this Pull Request:



PNPM 7 makes a lot of breaking changes to how commands are run, how dependencies are handled and how filters work. This is a broad set of changes to make all existing code work with pnpm 7.

Some example changes:

* For simplifying turbo commands we were working around issues in pnpm by adding `-- --` to commands. This isn't needed any more so we remove that in this PR

* `pnpm run --filter` used to implicitly be run recursively as far as I can tell (as if `-r` was passed). Now if you want to run a command recursively you need to pass `-r`. Also some commands in CI were using `--filter` where it would have sufficed to use `working-directory` to run the commands from their working directories.

* Remove all mentions of pnpm 6 from `engines` and CI

* Unfortunately now that peer deps are stricter in `pnpm 7` we have a lot of unresolved peer deps. Fixing this was insurmountable so I've disabled strictness. The warnings still show in CI and I hope teams will move forward with fixing these warnings as time goes on

* Fixed a few random TS errors that appeared after upgrading to pnpm 7

* Fix an issue where the code analyzer won't handle differences in pnpm version

And a few other bits and bobs!

### How to test the changes in this Pull Request:

1. See CI pass
2. Checkout branch locally, 
3. `rm -fr node_modules`, 
4. install pnpm 7 `npm i -g pnpm`, **please check the version is correct:** `pnpm -v`
5. `pnpm i`
6. `pnpm run build`

This should all work. Re-run build to see the cache get hit.

Run a pnpm command on a specific package either via -r or by working in the directory:

```
#option 1, use -r
pnpm run -r --filter="@woocommerce/date" build 

#option 2, use cd
cd packages/js/date
pnpm run build
```

### Other information:




### FOR PR REVIEWER ONLY:
